tooling/debugger/src/context.rs

```diff
@@ -21,6 +21,128 @@ use thiserror::Error;
 use std::collections::BTreeMap;
 use std::collections::{hash_set::Iter, HashSet};
 
+#[derive(Clone, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
+pub struct AddressMap {
+    // A Noir program is composed by
+    // `n` ACIR circuits
+    //       |_ `m` ACIR opcodes
+    //                |_ Acir call
+    //                |_ Acir Brillig function invocation
+    //                           |_ `p` Brillig opcodes
+    //
+    // We need to inline such structure to a one-dimension "virtual address" space
+    // We define the address space as a contiguous space where all ACIR and
+    // Brillig opcodes from all circuits are laid out
+    //
+    // `addresses: Vec<Vec<usize>>``
+    //  * The first vec is `n` sized - one element per ACIR circuit
+    //  * Each nested vec is `m` sized - one element per ACIR opcode in circuit
+    //    * Each element is the "virtual address" of such opcode
+    //
+    // For flattening we map each ACIR circuit and ACIR opcode with a sequential address number
+    // We start by assigning 0 to the very first ACIR opcode and then start accumulating by
+    // traversing by depth-first
+    //
+    //  * an ACIR call accumulates 1 (since it's a single opcode)
+    //  * an ACIR Brillig call accumulates as many opcodes the brillig function has (`p`)
+    //
+    // As a result the flattened addresses list may have "holes".
+    // This holes are a side effect of the brillig function calls
```

Comment:
It seems here you are treating ACIR calls as special, but they are like any other ACIR opcode. Only the `BrilligCall` opcode is replaced by the Brillig function's bytecode and expands in the address space, introducing "holes" in the otherwise sequential addresses of the consecutive ACIR opcodes. Maybe explain it in terms of what happens, not how (the phrase "accumulates as many opcodes" for example is an implementation detail if you like).